examples-gallery/beginner/plot_wheel_on_bumpy_road.py

```diff
@@ -0,0 +1,425 @@
+r"""
+Quarter Car Model on a Bumpy Road
+=================================
+
+Objective
+---------
+
+- Show on a simple example how to simultaneously optimize free
+  parameters and unknown trajectories of a system.
+- Show how one can iterate from a simpler problem (road less bumpy) to a
+  harder problem (road more bumpy).
+- Show how to use an additional state variable to get the acceleration of a
+  body into the objective function.
+- Show how to avoid possible unwanted unknown trajectories ( here
+  :math:`\dfrac{d^2}{dt^2}x_{car}`).
+
+
+Introduction
+------------
+
+A quarter car model must move from a starting point to a final point on a bumpy
+road. The body is connected to the wheel on the road by a linear spring and a
+damper.
+Body and wheel are modeled by particles. Movement is in the X/Z plane. Gravity
+points in the negative Z direction.
+
+I took the tire stiffness from here:
+
+https://kktse.github.io/jekyll/update/2021/07/18/re71r-255-40-r17-tire-vertical-stiffness.html
+
+
+**States**
+
+- :math:`x_{car}` : x position of the car [m]
+- :math:`z_{car}` : z position of the car [m]
+- :math:`z_{wheel}` : z position of the wheel [m]
+- :math:`ux_{car}` : x velocity of the car [m/s]
+- :math:`uz_{car}` : z velocity of the car [m/s]
+- :math:`uz_{wheel}` : z velocity of the wheel [m/s]
+- :math:`prevent_{jump}` : to ensure the wheel always touches the road [m]
+- :math:`steady_{body}` : to ensure that the vertical motions of the body are
+  not too large [m]
+- :math:`accel_{body}` : holds the acceleration of the body, which is to be
+  minimized [m/s^2]
+- :math:`accel_{street}` : holds the accelerations of the street as 'seen' by
+  the wheel. Only needed for the animation. [m/s^2]
+
+
+
+**Fixed Parameters**
+
+- :math:`m_{car}` : sprung mass [kg]
+- :math:`m_{wheel}` : unsprung mass [kg]
+- :math:`g` : gravity [m/s^2]
+- :math:`l_0` : equilibrium length of the spring [m]
+- :math:`r_1, r_2, r_3, r_4, r_5` : parameters of the street.
+
+
+**Free Parameters**
+
+- :math:`c` : damping constant [Ns/m]
+- :math:`k` : spring constant [N/m]
+- :math:`k1` : spring constant of the wheel [N/m]
+- :math:`l_{GW}` : equilibrium length of the wheel spring [m]
+
+
+**Unknown Trajectories**
+
+- :math:`f_x` : driving force [N]
+
+"""
```

Comment:
You need to explain somewhere that you are searching for the optimal suspension parameters to minimize the average acceleration felt by the sprung mass, given the road surface shape and whatever optimal propulsion you are choosing.